initializr-web/src/main/resources/static/js/start.js

```diff
@@ -156,12 +156,23 @@ $(function () {
             engine.add(data.dependencies);
         });
     };
+    var generatePackageName = function() {
```

Comment:
A package name cannot start with a number and the portion of package can't be a reserved keyword, that is

```
abstract	continue	for	new	switch
assert	default	goto	package	synchronized
boolean	do	if	private	this
break	double	implements	protected	throw
byte	else	import	public	throws
case	enum	instanceof	return	transient
catch	extends	int	short	try
char	final	interface	static	void
class	finally	long	strictfp	volatile
const	float	native	super	while
```

Those cases are probably super rare but I am wondering if we shouldn't identify them, let the `artifactId` be and just use `demo` as fallback package.

Similar rule should exist on the server side. 

I think the reserved part is a bit insance so we should probably stick with the numbers for now.